larsim/PhotonPropagation/PDFastSimPAR_module.cc

```diff
@@ -667,36 +818,105 @@ namespace phot {
+                                        fIsDomePDCorr,
+                                        fdelta_angulo_vuv,
+                                        fradius
+                                        );
+    if (!fIsFlatPDCorr && !fIsDomePDCorr) {
+      throw cet::exception("PDFastSimPAR")
+          << "Both isFlatPDCorr and isDomePDCorr parameters are false, at least one type of parameterisation is required for the semi-analytic light simulation." << "\n";
+    }
+    if (fIsFlatPDCorr) {
+      fPVS->LoadGHFlat(fGHvuvpars_flat,
+                       fborder_corr_angulo_flat,
+                       fborder_corr_flat
+                      );
+    }
+    if (fIsDomePDCorr) {
+      fPVS->LoadGHDome(fGHvuvpars_dome,
+                       fborder_corr_angulo_dome,
+                       fborder_corr_dome
+                      );
+    }
+
+    // Load corrections for VIS semi-analytic hits
+    if (fPVS->StoreReflected()) {
+      fStoreReflected = true;
+      std::cout << "Loading the reflected light corrections" << std::endl;
+      fPVS->LoadVisSemiAnalyticProperties(fdelta_angulo_vis,
+                                          fradius
+                                         );
+      if (fIsFlatPDCorr) {
+        fPVS->LoadVisParsFlat(fvis_distances_x_flat,
+                              fvis_distances_r_flat,
+                              fvispars_flat
+                            );
+      }
+      if (fIsDomePDCorr) {
+        fPVS->LoadVisParsDome(fvis_distances_x_dome,
+                              fvis_distances_r_dome,
+                              fvispars_dome
+                            );
+      }
+
+      // cathode dimensions
+      fcathode_ydimension = fActiveVolumes[0].SizeY();
+      fcathode_zdimension = fActiveVolumes[0].SizeZ();
+
+      // set cathode plane struct for solid angle function
+      fcathode_plane.h = fcathode_ydimension;
+      fcathode_plane.w = fcathode_zdimension;
+      fplane_depth = std::abs(fcathode_centre[0]);
+    }
+    else {
+      fStoreReflected = false;
+    }
   }
 
   //......................................................................
   // VUV semi-analytic hits calculation
-  int
-  PDFastSimPAR::VUVHits(const double Nphotons_created,
+  void
+  PDFastSimPAR::detectedDirectHits(std::map<size_t, int>& DetectedNumFast,
+                                   std::map<size_t, int>& DetectedNumSlow,
+                                   const double NumFast,
+                                   const double NumSlow,
+                                   geo::Point_t const& ScintPoint)
+  {
+    for (size_t const OpDet : util::counter(nOpChannels)) {
+      if (!isOpDetInSameTPC(ScintPoint, fOpDetCenter.at(OpDet))) continue;
+
+      // set detector struct for solid angle function
+      const PDFastSimPAR::OpticalDetector op{
+        fOpDetHeight.at(OpDet), fOpDetLength.at(OpDet),
+        fOpDetCenter.at(OpDet), fOpDetType.at(OpDet)};
```

Comment:
Is the `.at(OpDet)` necessary instead of `[OpDet]`?  `at` introduces additional bounds-checking that can be expensive when not necessary.